source/http/transport.go

```diff
@@ -0,0 +1,60 @@
+package http
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/upload"
+	"github.com/pkg/errors"
+)
+
+func newTransport(rt http.RoundTripper, sm *session.Manager, id string) http.RoundTripper {
+	return &sessionHandler{rt: rt, sm: sm, id: id}
+}
+
+type sessionHandler struct {
+	sm *session.Manager
+	rt http.RoundTripper
+	id string
+}
+
+func (h *sessionHandler) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.URL.Host != "buildkit-session" {
```

Comment:
What happens if there is a host named `buildkit-session`?

Can we use special URL scheme like `buildkit-session://` instead?

Reply:
> What happens if there is a host named buildkit-session?

This is why I used a name that should never collide. I can't use the ID and leave them unhandled because http source will also make `HEAD` requests that error and it would be unexpected if they actually start to dial now.

> Can we use special URL scheme like buildkit-session:// instead?

We could not use the existing frontends and buildkitd builds then as they hardcode http(s)

Reply:
@AkihiroSuda If you think we should use an even more obscure host, I'm fine with that. But it does show up in the output.